src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessInitialDataSetVersionFunctionTests.cs

```diff
@@ -97,4 +118,108 @@ public async Task ActivityFunctionThrowsException_CallsHandleFailureActivity()
             return mock;
         }
     }
+
+    public class HandleProcessingFailureTests(ProcessorFunctionsIntegrationTestFixture fixture)
+        : ProcessInitialDataSetVersionFunctionTests(fixture)
+    {
+        [Fact]
+        public async Task Success()
+        {
+            var instanceId = Guid.NewGuid();
+
+            DataSet dataSet = DataFixture.DefaultDataSet();
+
+            await AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion()
+                .WithDataSet(dataSet)
+                .WithStatus(DataSetVersionStatus.Processing)
+                .WithImports(
+                    () => DataFixture
+                        .DefaultDataSetVersionImport()
+                        .WithInstanceId(instanceId)
+                        .WithStage(DataSetVersionImportStage.Pending)
+                        .Generate(1)
+                )
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await AddTestData<PublicDataDbContext>(
+                context =>
+                {
+                    context.DataSetVersions.Add(dataSetVersion);
+                    context.DataSets.Update(dataSet);
+                }
+            );
+
+            var function = GetRequiredService<ProcessInitialDataSetVersionFunction>();
+            await function.HandleProcessingFailure(
```

Comment:
I was half-expecting to see that we'd be stashing an error of some kind on the import at this point?

Reply:
Replied in other related comment, thanks.